api/src/main/resources/liquibase-update-to-2.2.xml

```diff
@@ -0,0 +1,116 @@
+                </column>
+                <column name="changed_by" type="int" />
+                <column name="date_changed" type="datetime" />
+                <column name="retired" type="smallint" defaultValueNumeric="0">
+                    <constraints nullable="false" />
+                </column>
+                <column name="retired_by" type="int(11)" />
+                <column name="date_retired" type="datetime" />
+                <column name="retire_reason" type="varchar(255)" defaultValue="null" />
+                <column name="uuid" type="char(38)">
+                    <constraints nullable="false" unique="true" />
+                </column>
+            </createTable>
+            <addForeignKeyConstraint constraintName="program_attribute_type_creator_fk"
+                                     baseTableName="program_attribute_type" baseColumnNames="creator"
+                                     referencedTableName="users" referencedColumnNames="user_id" />
+            <addForeignKeyConstraint constraintName="program_attribute_type_changed_by_fk"
+                                     baseTableName="program_attribute_type" baseColumnNames="changed_by"
+                                     referencedTableName="users" referencedColumnNames="user_id" />
+            <addForeignKeyConstraint constraintName="program_attribute_type_retired_by_fk"
+                                     baseTableName="program_attribute_type" baseColumnNames="retired_by"
+                                     referencedTableName="users" referencedColumnNames="user_id" />
+        </changeSet>
+        <changeSet id="20151218-1530" author="Rahul,Swathi">
+            <preConditions onFail="MARK_RAN">
+                <not>
+                    <tableExists tableName="patient_program_attribute" />
+                </not>
+            </preConditions>
+             <comment>Creating patient_program_attribute table</comment>
+            <createTable tableName="patient_program_attribute">
+                <column name="patient_program_attribute_id" type="int" autoIncrement="true">
+                    <constraints primaryKey="true" nullable="false" />
+                </column>
+                <column name="patient_program_id" type="int"><constraints nullable="false" /></column>
+                <column name="attribute_type_id" type="int"><constraints nullable="false" /></column>
+                <column name="value_reference" type="text"><constraints nullable="false" /></column>
+                <column name="uuid" type="char(38)"><constraints nullable="false" unique="true" /></column>
+                <column name="creator" type="int"><constraints nullable="false"/></column>
+                <column name="date_created" type="datetime"><constraints nullable="false"/></column>
+                <column name="changed_by" type="int"/>
+                <column name="date_changed" type="datetime" />
+                <column name="retired_by" type="int(11)" />
```

Comment:
Since PatientProgramAttribute.hbm.xml is using voiding instead of retiring, let us use voiding here. voided, void reason, date voided, and voided by

Reply:
When done with the above, drop the two tables from your database and delete the corresponding rows from the liquibasechangelog table. This will let you run the changesets again to see if they are correct.

Reply:
Okay